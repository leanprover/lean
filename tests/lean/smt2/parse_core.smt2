(set-option :verbose true)
(set-option :use_locals)
(assert true)
(assert (not false))
(assert (and true true))
(assert (or true false))
(assert (xor true false))

(assert (= true true))
(assert (distinct true false))
(assert (not (distinct true false true)))
(assert (not (distinct true false true false)))
(assert (ite true true true))

(declare-sort X)
(declare-const x1 X)
(declare-const x2 X)
(declare-const x3 X)
(assert (= (ite true x1 x2) (ite false x2 x1)))
