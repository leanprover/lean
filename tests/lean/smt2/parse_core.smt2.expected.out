[assert] true
[assert] not false
[assert] and true true
[assert] or true false
[assert] xor true false
[assert] eq.{1} Prop true true
[assert] not (eq.{1} Prop true false)
[assert] not (and (and (not (eq.{1} Prop true false)) (not (eq.{1} Prop true true))) (not (eq.{1} Prop false true)))
[assert] not (and (and (and (and (and (not (eq.{1} Prop true false)) (not (eq.{1} Prop true true))) (not (eq.{1} Prop true false))) (not (eq.{1} Prop false true))) (not (eq.{1} Prop false false))) (not (eq.{1} Prop true false)))
[assert] ite.{1} true (classical.prop_decidable true) Prop true true
[declare_sort] X : Type
[declare_const] x1 : X
[declare_const] x2 : X
[declare_const] x3 : X
[assert] eq.{1} X (ite.{1} true (classical.prop_decidable true) X x1 x2) (ite.{1} false (classical.prop_decidable false) X x2 x1)
